app/models/concerns/current_tracked_by_changeset.rb

```diff
@@ -79,7 +79,19 @@ def instantiate_an_old_model
     end
 
     def changeable_attributes
-      @changeable_attributes ||= (self.attribute_names + @virtual_attributes - ['id', 'created_at', 'updated_at', 'created_or_updated_in_changeset_id', 'destroyed_in_changeset_id', 'version']).map(&:to_sym)
+      @changeable_attributes ||= (
+        self.attribute_names +
+        @virtual_attributes -
+        self.reflections.values.map(&:foreign_key) -
+        [
+          'id',
+          'created_at',
+          'updated_at',
+          'created_or_updated_in_changeset_id',
+          'destroyed_in_changeset_id',
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.